cmd/nodecmd/create.go

```diff
@@ -694,124 +692,100 @@ func createNodes(cmd *cobra.Command, args []string) error {
-					nodeResults.AddResult(monitoringHost.NodeID, nil, err)
+					nodeResults.AddResult(monitoringHost.IP, nil, err)
 					ux.SpinFailWithError(spinner, "", err)
 					return
 				}
 				ux.Logger.Info("RunSSHSetupPrometheusConfig completed")
 				if err := ssh.RunSSHSetupLokiConfig(monitoringHost, constants.AvalanchegoLokiPort); err != nil {
-					nodeResults.AddResult(monitoringHost.NodeID, nil, err)
+					nodeResults.AddResult(monitoringHost.IP, nil, err)
 					ux.SpinFailWithError(spinner, "", err)
 					return
 				}
 				ux.Logger.Info("RunSSHSetupLokiConfig completed")
 				if err := docker.ComposeSSHSetupMonitoring(monitoringHost); err != nil {
-					nodeResults.AddResult(monitoringHost.NodeID, nil, err)
+					nodeResults.AddResult(monitoringHost.IP, nil, err)
 					ux.SpinFailWithError(spinner, "", err)
 					return
 				}
 				ux.Logger.Info("ComposeSSHSetupMonitoring completed")
 				ux.SpinComplete(spinner)
 			}(&wgResults, monitoringHost)
 		}
+		wg.Wait()
+		spinSession.Stop()
 	}
 	for _, host := range hosts {
-		wg.Add(1)
-		go func(nodeResults *models.NodeResults, host *models.Host) {
-			defer wg.Done()
-			if err := host.Connect(0); err != nil {
-				nodeResults.AddResult(host.NodeID, nil, err)
-				return
-			}
-			if err := provideStakingCertAndKey(host); err != nil {
-				nodeResults.AddResult(host.NodeID, nil, err)
-				return
-			}
-			spinner := spinSession.SpinToUser(utils.ScriptLog(host.NodeID, "Setup Node"))
-			if err := ssh.RunSSHSetupNode(host, app.Conf.GetConfigPath()); err != nil {
-				nodeResults.AddResult(host.NodeID, nil, err)
-				ux.SpinFailWithError(spinner, "", err)
-				return
-			}
-			if err := ssh.RunSSHSetupDockerService(host); err != nil {
-				nodeResults.AddResult(host.NodeID, nil, err)
-				ux.SpinFailWithError(spinner, "", err)
-				return
-			}
-			ux.SpinComplete(spinner)
-			if addMonitoring {
-				cloudID := host.GetCloudID()
-				nodeID, err := getNodeID(app.GetNodeInstanceDirPath(cloudID))
-				if err != nil {
-					nodeResults.AddResult(host.NodeID, nil, err)
-					ux.SpinFailWithError(spinner, "", err)
-					return
-				}
-				if err = ssh.RunSSHSetupPromtailConfig(host, monitoringNodeConfig.PublicIPs[0], constants.AvalanchegoLokiPort, cloudID, nodeID.String(), ""); err != nil {
-					nodeResults.AddResult(host.NodeID, nil, err)
-					ux.SpinFailWithError(spinner, "", err)
-					return
+		publicAccessToHTTPPort := slices.Contains(cloudConfigMap.GetAllAPIInstanceIDs(), host.GetCloudID()) || publicHTTPPortAccess
+		host.APINode = publicAccessToHTTPPort
+	}
+	if err = setup(hosts, avalancheGoVersion, network); err != nil {
+		return err
+	}
+	if addMonitoring {
+		spinSession := ux.NewUserSpinner()
+		for _, host := range hosts {
+			wg.Add(1)
+			go func(nodeResults *models.NodeResults, host *models.Host) {
+				defer wg.Done()
+				spinner := spinSession.SpinToUser(utils.ScriptLog(host.IP, "Add Monitoring"))
+				if addMonitoring {
+					cloudID := host.GetCloudID()
```

Comment:
we should remove host.GetCloudID() and change logic behind RunSSHSetupPromtailConfig and 
GetNodeInstanceDirPath(cloudID)